Python/bytecodes.c

```diff
@@ -465,6 +465,20 @@ dummy_func(
 
         // stack effect: (__0 -- )
         inst(BINARY_SUBSCR) {
+            if (!cframe.use_tracing) {
```

Comment:
This bothers me.
First of all the extra branch might slow things down, but `DO_TRACING` was reasonably self contained and `cframe.use_tracing` didn't need to checked too many other places.

Would incrementing the counter in `DO_TRACING`, so that `ADAPTIVE_COUNTER_IS_ZERO(cache->counter)` is guaranteed to be false and the `DECREMENT_ADAPTIVE_COUNTER` is cancelled out?
In `DO_TRACING` add something like?
```C
    if (is_adaptive(opcode)) {
        INCREMENT_ADAPTIVE_COUNTER(next_instr);
    }
```

Reply:
It's a tiny bit trickier (we have to make sure we don't overflow the counter to zero), but sure, I'll try that!